### ISSUE TITLE
Code Of Conduct solves a non-problem, creates risk.

### ISSUE THREAD

**Issue by mryellow** [QubesOS/qubes-issues#4377]
I was under the impression you had not kowtowed to this, as the file had not been added to your repositories. Now I see this is not the case and you have exposed yourself (and your users) to this risk.

The Code Of Conduct published on your [website](https://www.qubes-os.org/code-of-conduct/) is designed to curtail speech and enable expulsion of core contributors based on petty personal and/or political [Tone Policing](https://en.wikipedia.org/wiki/Tone_policing).

Please consider using something along these lines instead:

# Code of Conduct

## Free Speech and Consideration

Everyone has the right to communicate opinions and ideas without fear of censorship. Consideration shall be given to all constructive speech that is based in fact and reason.

**Comment by andrewdavidwong** [comment]
I've closed this because it's highly unlikely to result in any productive discussion or change given the politically-charged nature of the issue description. The objections to the Code of Conduct presented here are vague and unsubstantiated. The main objection seems to be that we're endorsing "tone policing," which is simply false, as anyone can see from the discussion in #2163, where I specifically argued against forcing anyone to exhibit "good behavior."

Having said this, we're always open to revising our Code of Conduct for legitimate reasons. For example, if particular language in the Code of Conduct is ambiguous or fails to accurately convey our intent, that would be a good reason to change it.

**Comment by mryellow** [comment]
> politically-charged nature 

Exactly. You've exposed yourself to this risk. Not saying anything would be safer.

>  specifically argued against forcing anyone to exhibit "good behavior."

Thankyou.

**Comment by mryellow** [comment]
Must say it's pleasing to see you've actually thought about this before blindly including it. Most projects have simply one-click installed a bunch of politically loaded words into their contracts without a second thought.

There is a point made in #2163 that it should be explicitly stated that this is a modified version. As it stands it appears much the same as the form-letter version included in other projects.

Perhaps removing the link to the politically loaded `contributor-covenant.org` would be a good additional change and remove some of this taint.

**Comment by andrewdavidwong** [comment]
>> politically-charged nature
>>
> Exactly. You've exposed yourself to this risk. Not saying anything would be safer.

I was referring to the politically-charged nature of your issue description. It would not necessarily be safer not to have a Code of Conduct at all, since then it would be more difficult to deal with genuine troublemakers. Such a person could argue that we were acting arbitrarily or based on personal prejudices, since we could not point to an established public set of guidelines that the offending individual had violated.

> Must say it's pleasing to see you've actually thought about this before blindly including it. Most projects have simply one-click installed a bunch of politically loaded words into their contracts without a second thought.

Yes, we take this stuff seriously. We strive to say what we mean and mean what we say.

> There is a point made in #2163 that it should be explicitly stated that this is a modified version. As it stands it appears much the same as the form-letter version included in other projects.

I think a productive use of this issue would be to add an introduction to the Code of Conduct explaining the purpose of the document, what it is intended to be, and what it is not intended to be. I think this should head off similar misunderstandings in the future. I'll add this momentarily.

> Perhaps removing the link to the politically loaded `contributor-covenant.org` would be a good additional change and remove some of this taint.

That's just an attribution link. It's simply saying that our Code of Conduct is adapted from that one, among others. It's not an endorsement of that Code of Conduct or organization. Removing the link would not be appropriate, since then we would not be citing a source that we used.

**Comment by mryellow** [comment]
> I was referring to the politically-charged nature of your issue description.

Would this issue exist if the project hadn't opened the door for these types of politics?

This is exactly my problem with these documents, they are unneeded and rather than solving issues, they create them.

> That's just an attribution link.

Yes, it attributes the words to a politically charged organisation which is working to empower itself and then further promotes them with a link.

**Comment by andrewdavidwong** [comment]
> Would this issue exist if the project hadn't opened the door for these types of politics?

Yes, because the free expression issue arose independently before our Code of Conduct existed, and there will always be conduct-related disputes in any large, diverse community of people. The lack of a Code of Conduct exposes the project to a greater risk than having one, as I explained above.

> This is exactly my problem with these documents, they are unneeded and rather than solving issues, they create them.

That's not accurate. They are needed for the reasons explained above. They prevent more problems than they create. You're only consider the potential problems arising from the existence of the Code of Conduct. You're not considering all of the problems that it prevents or makes it easy to solve. Just as you're complaining about the existence of the Code of Conduct, people would have complained about the absence of one. We can't please everyone. All we can do is try our best to be reasonable and practical.

> Yes, it attributes the words to a politically charged organisation which is working to empower itself and then further promotes them with a link.

Citing a source is not promoting or endorsing the source. Even when writing a critical article, for example, it is standard practice to cite the source being criticized. It would be an act of plagiarism to use a source without citing it, and we would never intentionally do that.

I believe the productive component of this discussion has run its course. We have taken reasonable action to address your concerns with the Code of Conduct. The remaining disagreement is a fundamentally political one that lies beyond the scope of a qubes-issue. This is not an appropriate venue for that kind of political or philosophical debate.

**Comment by mryellow** [comment]
Another potential approach for your reference.

https://www.sqlite.org/codeofconduct.html
